executor/write.go

```diff
@@ -1366,6 +1357,39 @@ func (e *UpdateExec) Next(goCtx goctx.Context) (Row, error) {
 	return Row{}, nil
 }
 
+// Next implements the Executor Next interface.
+func (e *UpdateExec) Next(goCtx goctx.Context) (Row, error) {
+	if !e.fetched {
+		err := e.fetchRows(goCtx)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		e.fetched = true
+	}
+
+	return e.exec(goCtx, e.SelectExec.Schema())
+}
+
+// NextChunk implements the Executor NextChunk interface.
+func (e *UpdateExec) NextChunk(goCtx goctx.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+	if !e.fetched {
+		err := e.fetchChunkRows(goCtx)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		e.fetched = true
+	}
+
+	row, err := e.exec(goCtx, e.children[0].Schema())
+	if row != nil && err == nil {
+		// there is more rows to exec.
+		// TODO: exactly it is not a error, but there is no graceful way to indicate there is more rows.
+		return ErrMoreRows
```

Comment:
you can wrap a `for` loop for line 1384~1390, thus `ErrMoreRows` can be removed.